src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsStateReset.hpp

```diff
@@ -0,0 +1,110 @@
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file SolidMechanicsStateReset.hpp
+ */
+
+#ifndef SRC_CORECOMPONENTS_PHYSICSSOLVERS_SOLIDMECHANICS_SOLIDMECHANICSSTATERESET_HPP_
+#define SRC_CORECOMPONENTS_PHYSICSSOLVERS_SOLIDMECHANICS_SOLIDMECHANICSSTATERESET_HPP_
+
+#include "events/tasks/TaskBase.hpp"
+
+namespace geosx
+{
+
+class SolidMechanicsLagrangianFEM;
+
+/**
+ * @class SolidMechanicsStateReset
+ *
+ * Task class allowing for various "reset" actions on state variables, useful for pre-
+ * and post-equilibration actions. Currently, this class gives the user the option
+ * to reset displacements to zero and enable/disable inelastic behavior. A common
+ * workflow might be:
+ * (1) resetDisplacments = false, disableInelasticity = true,
+ * (2) solve an equilibration problem for initial conditions,
+ * (3) resetDisplacements = true, disablePlasticity = false,
+ * (4) run a normal simulation
+ *
+ */
+class SolidMechanicsStateReset : public TaskBase
+{
+public:
+
+  /**
+   * @brief Constructor for the state reset class
+   * @param[in] name the name of the task coming from the xml
+   * @param[in] parent the parent group of the task
+   */
+  SolidMechanicsStateReset( const string & name,
+                            Group * const parent );
+
+  /// Destructor for the class
+  ~SolidMechanicsStateReset() override;
+
+  /// Accessor for the catalog name
+  static string catalogName() { return "SolidMechanicsStateReset"; }
+
+  /**
+   * @defgroup Tasks Interface Functions
+   *
+   * This function implements the interface defined by the abstract TaskBase class
+   */
+  /**@{*/
+
+  virtual bool execute( real64 const time_n,
+                        real64 const dt,
+                        integer const cycleNumber,
+                        integer const eventCounter,
+                        real64 const eventProgress,
+                        DomainPartition & domain ) override;
+
+  /**@}*/
+
+private:
+
+  /**
+   * @struct viewKeyStruct holds char strings and viewKeys for fast lookup
+   */
+  struct viewKeyStruct
+  {
+    /// String for the solid solver name
+    constexpr static char const * solidSolverNameString() { return "solidSolverName"; }
+    /// String for the displacement reset flag
+    constexpr static char const * resetDisplacementsString() { return "resetDisplacements"; }
+    /// String for the inelasticity flag
+    constexpr static char const * disableInelasticityString() { return "disableInelasticity"; }
+  };
+
+
+  void postProcessInput() override;
+
+  /// Name of the solid mechanics solver
+  string m_solidSolverName;
+
+  /// Pointer to the solid mechanics solver
+  SolidMechanicsLagrangianFEM * m_solidSolver;
+
+  /// Flag for resetting displacements (and velocities)
+  int m_resetDisplacements;
+
+  /// Flag for disabling / enabling inelasticity
+  int m_disableInelasticity;
```

Comment:
It is six-of-one, half-a-dozen of the other.  We would use `!m_isInelasticityEnabled` almost everywhere in the code, so I went with the reverse.

Reply:
Yes I understand, it's just that this wording bothers me a little.

This boolean is meant to reflect a "_status_" stored: "_Is the inelasticity disabled? Yes or no._".
"_Disable inelasticity_" is the "_action_" that lead to the "_status_" of inelasticity.

Like the name of the member function: `solver.disableInelasticity( bool )` sounds weird to me:
You may want to "_disable inelasticity_", but the action may be inverted depending on the `boolean` flag.

Whatever, I'm surely overthinking a detail here, it's not important.

Reply:
Ah, I see.  Okay, what about `solver.setInelasticityFlag( bool )`?  Since this is a public function we want a clear name.

I would probably leave the `m_disableInelasticity` flag as is, though, as I think the usage is obvious in context: `if( m_disableInelasticity ) { ... }`.  Do we have a common convention for bool flags in the code? `m_flagDisableInelasticity` or `m_isInelasticityDisabled` all sound fine to me, but not much better than what it is now.

Reply:
Sorry, my previous suggestion is terrible and will make things more confusing.  How about we have two separate functions `enableInelasticity()` and `disableInelasticity()` and just put switch in the Reset class?

Reply:
Something like `solver.setInelasticity( bool )` (with or without the `Flag` suffix) makes immediate sense I think.

The `enableInelasticity`/`disableInelasticity` can be nice too; it mostly depends on how we want to use it. Having the client code repeatedly do some thing like `if( flag ) { solver.enableInelasticity() } else { solver. disableInelasticity() }` can be heavy.

Regarding the boolean convention, I guess we do not have any strong requirement.
A common rule that I do personally like is to start any boolean with `is` or `has` because the meaning of the boolean is often non ambiguous that way.
`m_isInelasticityDisabled` leaves no room for interpretation. (Or maybe `m_isInelasticityEnabled` to prevent from double negations like `!m_isInelasticityDisabled` which means `enabled` 😉 )

But it's also context and habits dependent, let's not create problems where there is none!